libs/model/test/referral/referral-lifecycle.spec.ts

```diff
@@ -0,0 +1,66 @@
+import { Actor, command, dispose } from '@hicommonwealth/core';
+import { models } from '@hicommonwealth/model';
+import { ChainNode } from '@hicommonwealth/schemas';
+import { ChainBase, ChainType } from '@hicommonwealth/shared';
+import { afterAll, beforeAll, describe, expect, it } from 'vitest';
+import { z } from 'zod';
+import { CreateCommunity } from '../../src/community';
+import { CreateReferralLink } from '../../src/user';
+import { seedCommunity } from '../utils/community-seeder';
+
+describe('Referral lifecycle', () => {
+  let admin: Actor;
+  let member: Actor;
+  let node: z.infer<typeof ChainNode>;
+
+  beforeAll(async () => {
+    const { node: _node, actors } = await seedCommunity({
+      roles: ['admin', 'member'],
+    });
+    admin = actors.admin;
+    member = actors.member;
+    node = _node!;
+  });
+
+  afterAll(async () => {
+    await dispose()();
+  });
+
+  it('should create a referral', async () => {
+    // admin creates a referral link
+    const response = await command(CreateReferralLink(), {
+      actor: admin,
+      payload: {},
+    });
+
+    // member creates a community using the referral link
+    const id = 'test-community-with-referral-link';
+    await command(CreateCommunity(), {
+      actor: member,
+      payload: {
+        chain_node_id: node.id!,
+        id,
+        name: id,
+        type: ChainType.Offchain,
+        base: ChainBase.Ethereum,
+        default_symbol: 'TEST',
+        social_links: [],
+        directory_page_enabled: false,
+        tags: [],
+        referral_link: response?.referral_link,
+      },
+    });
+
+    // drain the outbox
```

Comment:
@timolegros do we have something like this? or should I create the util?

Reply:
No, but is a util here really necessary? Seems like the util would just replicate the RabbitMQ routing config/behaviour in which case you end up testing the test util which is pointless. Isn't it easier to just separate into 2 steps:
1. Check if event was emitted to Outbox
2. Take the event from the outbox and explicitly pass it to the projection handler.

Reply:
No, the util should do what you just described in 2 steps... just to encapsulate that